### ISSUE TITLE
TensorBoard Feature Request: [DASHBOARD] Static Variable Display Tab

### ISSUE THREAD

**Issue by Mazecreator** [tensorflow/tensorflow#4714]
I do a lot of different runs which contain descriptions as well as different meta parameters.  It would be nice to have a tab on TensorBoard so I can tell which experiment I am viewing.

My thought would be to simply add static variables from the Graph which might include Strings, Ints, Floats, (maybe matrix in table format if possible, like PCA parameters, maybe confusion matrix, etc).  These may include a text description, current learning rate, network meta parameters, etc.  It would be good to simple present the "current" value of these variables so if there is a search on Discount Rates for RL or variation such as learning rate, those can be viewed.  This would be different from simply graphing the Learning rate as that shows the history rather than "current".  I think of this as a TensorFlow "Dashboard" that I can setup.

Something like:
tf.dashboard_summary(tags, tensor/value, collections=None, name=None)

eg:
tf.dashboard_summary("Description:",description_string_tensor_or_python_string)
tf.dashboard_summary("Learning Rate:",lr_tensor)
tf.dashboard_summary("DiscountRate:",discountRate_python_variable)

In this case, a "DASHBOARD" tab on tensorflow would contain the 3 labels above and the tensor/value.  If the value is a python variable, then it should be considered constant and will not change over the graph lifecycle.  If it is a tensor/variable, then it should be pulled from the graph at each iteration.

I can help but not sure where to jump in to get this started.

**Comment by Mazecreator** [comment]
Another addition that might be nice to add would be a COLLECTIONS tab.  Simply display the names of the Operation & Tensors within all the collections.  This should include Tensor Size, etc, whatever can be extracted from the graph.  This would be a quick reverence to the topology of the graph and helpful if you go back to a different structure and looking to see what the Network looks like or what the input variable names are for Serving, etc.

**Comment by Mazecreator** [comment]
Is there anything I can do to help with this request?

**Comment by danmane** [comment]
Sorry for the slow response, and thanks for the suggestion. 
I have some questions:
1. The "dashboard" tab - would it show one run at a time (like the graph page) or would it be showing multiple runs simultaneously?
2. It sounds like some of the information it shows would be constant for the run (e.g. description) and some would be variable (e.g. learning rate). So we could just re-use the existing data input for the variable ones, like the learning rate, right? But we would need to add some metadata to the summary so we know that we want it to display in the dashboard tab. 
3. Can you maybe submit an image of a mock up, so I have a clearer idea of what you have in mind? A hand-drawn sketch would be fine.

This is pretty well-aligned with features and data sources we've wanted to add to TensorBoard. If you're interested in working on this we can discuss in a lot more detail the work that will go into this (maybe with a video chat). Do you have any frontend development experience / are you interested in getting your hands dirty with frontend development?

**Comment by Mazecreator** [comment]
1) I am not that familiar with multiple runs as I haven't managed to get that working.  But my feeling would be to "select" the visible run and if un-selected use the "latest" run to show.  For me, a single run is fine as I am trying to find the meta-parameters and a description of the experiment as I have a few running in parallel and it takes time to figure out which is which.

2) That is correct, it might be good to "offer" collections of data such as weights, gradients, etc.  These would be the "latest" values, and it would be good to view them in the browser if small in size/clipped to a max to maintain performance, and offer a "csv" export of the variables for excel or some other means to view.  It would be good if possible to render a 2D image of the variable for 2D data to help visualize what is going on within the graph.

3) I am hoping to get TensorBoard running again soon as it broke a while ago to get a screen to do the mock-up.  My understanding is there is a roll-back in progress which should fix this for now.  I will work on that once I get the fix.

It has been 5 years or so since I have worked on HTML so a little rusty on getting the data to look nice in a web browser, but I can help get the data that far (and make a stab at it).  My feeling is there is a lot of data stored in the logs, maybe access "check points" as well to get real-time training data.

**Comment by Mazecreator** [comment]
I made a first shot at the DASHBOARD page.  The text in the summary should be aligned but Paint didn't help me there so I drew a gray line to show they are all aligned (line optional).

Hopefully this give you a good idea of what is in my mind with this request.  Tensors might be exportable or can be viewed in a grid on a separate page if they are larger than a certain size (for performance).  Rank 1 and 2 tensors should be able to be viewed & exported this way.  The same feature should exist for the COLLECTION browser.  I didn't see a gradient collection so that might need to be added manually to the collection.

![dashboard concept](https://cloud.githubusercontent.com/assets/18412448/19873583/3614a74a-9f96-11e6-82b8-9495883f33e7.png)

**Comment by Mazecreator** [comment]
HI @danmane,

Just checking back in to see if the image above helped you with my thoughts on the "Dashboard" approach.

Regards,
Greg

**Comment by danmane** [comment]
Hey greg, thanks for the mock up.

This does look like a good fit for the TensorBoard plugin system, and for some additional run metadata that we are planning to add that could populate the Description, Revision, Topology, fields, etc. I could imagine something like this being the default starting page for TensorBoard.

However, on our end we won't be ready to really start integrating plugins like this until the end of January at the earliest. Do you mind if we put this issue on hold for a few months while we get things ready on the TensorBoard side?

**Comment by Mazecreator** [comment]
Hi @danmane ,

That is fine, I just wanted to make sure the topic didn't get dropped due to lack of interest.  I will have more time in January as well if you want some help.

**Comment by aselle** [comment]
Ping @dandelionmane.

**Comment by liorshk** [comment]
+1

**Comment by dandelionmane** [comment]
Migrating this to the TensorBoard repo.
